src/statsmodels/formula.jl

```diff
@@ -283,6 +287,20 @@ end
 
 const DEFAULT_CONTRASTS = DummyCoding
 
+_levels(x::Union{CategoricalArray, NullableCategoricalArray}) = levels(x)
+
+function _levels{T<:Nullable}(x::AbstractArray{T})
+    levs = [get(l) for l in unique(x) if !isnull(l)]
+    try; sort!(levs); end
```

Comment:
Because not all types support `isless`. And `method_exists` is actually not enough since it doesn't work with `Array{Any}`, and `isless` might even fail when it's defined (e.g. with unordered `CategoricalValue`).